src/main/webapp/app/grading-system/grading-system.component.ts

```diff
@@ -118,12 +120,95 @@ export class GradingSystemComponent implements OnInit {
         }
     }
 
+    /**
+     * Checks if the currently entered grade steps are valid based on multiple criteria:
+     * - there must be at least one grade step
+     * - all fields must be filled out
+     * - the percentage values must lie between 0 and 100 (both inclusive)
+     * - all grade names must be unique
+     * - the first passing must be set if the scale is of GRADE type
+     * - the bonus points are at least 0 if the scale is of BONUS type
+     * - the bonus points must be strictly ascending in values
+     * - the max and min % of adjacent grade steps overlap
+     * - the first grade step begins at 0% and the last ends at 100%
+     */
+    validGradeSteps(): boolean {
+        if (!this.gradingScale || this.gradingScale.gradeSteps.length === 0) {
+            this.invalidGradeStepsMessage = this.translateService.instant('artemisApp.gradingSystem.error.empty');
+            return false;
+        }
+        // check if any of the fields are empty
+        for (const gradeStep of this.gradingScale.gradeSteps) {
+            if (gradeStep.gradeName === '' || gradeStep.gradeName === null || gradeStep.lowerBoundPercentage === null || gradeStep.upperBoundPercentage === null) {
+                this.invalidGradeStepsMessage = this.translateService.instant('artemisApp.gradingSystem.error.emptyFields');
+                return false;
+            }
+        }
+        // check if any of the fields have invalid percentages
+        for (const gradeStep of this.gradingScale.gradeSteps) {
+            if (gradeStep.lowerBoundPercentage < 0 || gradeStep.upperBoundPercentage > 100 || gradeStep.lowerBoundPercentage >= gradeStep.upperBoundPercentage) {
+                this.invalidGradeStepsMessage = this.translateService.instant('artemisApp.gradingSystem.error.invalidMinMaxPercentages');
+                return false;
+            }
+        }
```

Comment:
Because I would like to check the first condition for all steps first, and the check it for the second one. It's a lot more consistent with the warnings.